bin/fetch-configlet

```diff
@@ -1,16 +1,26 @@
 #!/usr/bin/env python3.7
+from datetime import datetime, timezone
 import io
-import tarfile
+import json
 import platform
-import requests
 import sys
+import tarfile
+import time
+from urllib.request import urlopen
 
 
 def download_and_extract(url):
-    resp = requests.get(url)
-    print('extracting...')
-    file = tarfile.open(fileobj=io.BytesIO(resp.content), mode='r:gz')
-    file.extractall(path='bin/')
+    with urlopen(url) as request:
+        status = request.status
+        if status >= 400:
+            print(f"Unexpected {status} status from download server.")
+            time.sleep(5)
+            return 1
+        response = request.read()
+    print("extracting...")
+    file = tarfile.open(fileobj=io.BytesIO(response), mode="r:gz")
```

Comment:
Not that it's very impactful, but do we need to specify the compression mode? I believe `mode="r"` or `mode=r:*` will handle gzip compression transparently, while specifying `mode="r:gz" will fail were Github to ever decide to switch to one of the other **tarfile** supported compression schemes.